src/App.tsx

```diff
@@ -1,26 +1,169 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState, useRef } from 'react';
+import { getTodos, deleteTodo, addTodo } from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { Todo } from './types/Todo';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TempTodo } from './components/TempTodo';
 
-const USER_ID = 0;
+export enum TypeFilter {
+  All = 'All',
+  Active = 'Active',
+  Completed = 'Completed',
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todosList, setTodosList] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState('All');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodo, setLoadingTodo] = useState<number[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (!isLoading) {
+      inputRef.current?.focus();
+    }
+  }, [isLoading]);
+
+  const listOfActiveTodos = todosList.filter(todo => !todo.completed);
+  const hasActiveTodo = todosList.some(todo => todo.completed);
+
+  useEffect(() => {
+    if (!errorMessage) {
+      return;
+    }
+
+    const timeoutId = setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+
+    return () => {
+      clearTimeout(timeoutId);
+    };
+  }, [errorMessage]);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodosList)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
+  }, []);
+
+  function deletePost(postId: number): Promise<void> {
+    setLoadingTodo(prev => [...prev, postId]);
+
+    return deleteTodo(postId)
+      .then(() => {
+        setIsLoading(true);
+        setTodosList(todos => todos.filter(todo => todo.id !== postId));
+      })
+      .catch(() => setErrorMessage('Unable to delete a todo'))
+      .finally(() => {
+        setIsLoading(false);
+        setLoadingTodo(prev => prev.filter(id => id !== postId));
+        inputRef.current?.focus();
+      });
+  }
+
+  const deleteCompletedTasks = () => {
+    const completedTodos = todosList.filter(todo => todo.completed);
+
+    setLoadingTodo(completedTodos.map(todo => todo.id));
+
+    Promise.all(completedTodos.map(todo => deletePost(todo.id))).finally(() => {
+      setLoadingTodo([]);
+    });
+  };
+
+  async function addPost(newTodo: Omit<Todo, 'id'>): Promise<void> {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const addPost = async (newTodo: Omit<Todo, 'id'>): Promise<void> => {
```